### PR TITLE
move  from daemon_restart.go to framework/util.go

### PR DESCRIPTION
**Pull request by jeffvance**
**What this PR does / why we need it**:
Moves the func `nodeExec` from daemon_restart.go to framework/util.go. This is the correct file for this func and is a more intuitive pkg for other callers to use. This is a small step of the larger effort of restructuring e2e tests to be more logically structured and easier for newcomers to understand.

```release-note
NONE
```
cc @timothysc @copejon

### CODE REVIEW

**Comment by jeffvance** [comment]
@k8s-bot test this

**Comment by jeffvance** [comment]
@k8s-bot bazel test this

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -3835,7 +3844,7 @@ func IssueSSHCommandWithResult(cmd, provider string, node *v1.Node) (*SSHResult,
 	host := ""
 	for _, a := range node.Status.Addresses {
 		if a.Type == v1.NodeExternalIP {
-			host = a.Address + ":22"
+			host = fmt.Sprintf("%v:%v", a.Address, sshPort)
```
Comment (timothysc) [inline test/e2e/framework/util.go]: seems orthogonal to your change.

Comment (jeffvance) [inline test/e2e/framework/util.go]: It is. But using Sprintf vs. string addition is a more common pattern from what I've seen, and I was on that line of code. I can change it back if it matters.

**Comment by jeffvance** [comment]
@liggitt @timothysc  I can't discover the bazel build failure. All I see is this (each time I've re-run):
`Unable to load build details from gs://kubernetes-jenkins/pr-logs/pull/45423/pull-kubernetes-bazel/16179`

**Comment by ixdy** [comment]
Issue with prow.
@k8s-bot bazel test this

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -101,6 +101,8 @@ import (
 	testutils "k8s.io/kubernetes/test/utils"
 )
 
+const sshPort = "22"
```
Comment (copejon) [inline test/e2e/framework/util.go]: push this down to `const ( ... )` block on L106?

**Comment by jeffvance** [comment]
@k8s test this

**Comment by jeffvance** [comment]
@k8s-bot unit test this

**Comment by jeffvance** [comment]
@liggitt @saad-ali can I get a quick review and an approve? thanks

**Review by liggitt — COMMENTED** [review]
net.JoinHostPort instead of sprintf

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -3795,6 +3797,13 @@ type SSHResult struct {
 	Code   int
 }
 
+// NodeExec execs the given cmd on node via SSH. Note that the nodeName is an sshable name,
+// eg: the name returned by framework.GetMasterHost(). This is also not guaranteed to work across
+// cloud providers since it involves ssh.
+func NodeExec(nodeName, cmd string) (SSHResult, error) {
```
Comment (liggitt) [inline test/e2e/framework/util.go]: Is this a move? I don't see it being removed from somewhere else

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -3795,6 +3797,13 @@ type SSHResult struct {
...
+	return SSH(cmd, fmt.Sprintf("%v:%v", nodeName, sshPort), TestContext.Provider)
```
Comment (liggitt) [inline test/e2e/framework/util.go]: Not thrilled with exporting package functions that depend on global vars

**Comment by saad-ali** [comment]
I will let @liggitt review this. Once he's happy with it :+1:

File: test/e2e/framework/util.go (hunk 3 above)
Comment (jeffvance) [inline test/e2e/framework/util.go]: Duh! Thanks. Somehow daemon_restart.go didn't get commited.

File: test/e2e/framework/util.go (hunk 4 above)
Comment (jeffvance) [inline test/e2e/framework/util.go]: Ok. I'm open to something better. I see this type of pattern where an exported func calls an internal func, adding args to the call. Not sure if the added args are global or not. Is this a deal breaker?

Comment (liggitt) [inline test/e2e/framework/util.go]: I won't block on it, but it seems like increased surface area without a whole lot of benefit

**Comment by liggitt** [comment]
go ahead and squash to a single commit, then LGTM

**Comment by jeffvance** [comment]
@k8s-bot unit test this

**Comment by jeffvance** [comment]
Thank you @liggitt. I need an /approve and /lgtm.

**Comment by jeffvance** [comment]
@k8s-bot test this

**Comment by liggitt** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by jeffvance** [comment]
@k8s-bot unit test this
@k8s-bot pull-kubernetes-federation-e2e-gce test this
